src/test/java/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -190,6 +190,12 @@ public void throwsParseExceptionOnBadRangeForm() throws IOException {
         parse(r("bad-range-form.xml"));
     }
 
+    @Test
+    public void formWithCountNonEmptyFunc_ShouldNotThrowException() throws IOException {
```

Comment:
I'm not sure what's the goal of this test. Could you elaborate? 

I think I was expecting some kind of assertion here. Maybe assert that after creating some repeats those nodes with the `count-non-empty` and `count` calculate expressions have some expected values.

Reply:
We don't need any assertion here because without the fix the form just can't be parsed and an exception is thrown:
`java.lang.RuntimeException: Error evaluating field 'weight_mean2': unrecognized data type in 'calculate' expression: java.lang.Integer`
then fails. With the fix it doesn't fail so I think it's enough.

Reply:
What I've done in similar cases is asserted that the form title matches expectations. That way it's really a test to make sure that the form parse worked.

Reply:
I'm sorry to be nitpicky, but that's not how tests work... A test should help us document some behavior and/or prevent regressions. Neither things are provided by this test.

I think it's a missed opportunity to explain what the `count` and `count-non-empty` functions do in the context of a form. Adding some should be easy with the tooling we have in place.

If I can't seduce you into doing this, I'd rather remove the test.

Reply:
@ggalmazor I think it’s essential that we have a test that parses a form with the function call. It’s the lack of such a test that led to the broken behavior. This test as written would crash on master. I agree that not having any assertion is sub-optimal but I don’t see what’s so bad about doing an assertion that confirms that the form parse was successful. 

The behavior of the function is tested in XPathEvalTest. I don’t see the benefit of testing it again here. It would require filling in values and I think it would obscure the true goal of the test which is really ensuring that the function is properly parsed. That said, I think we’ve collectively spent too much time on this already and if you feel strongly about it’s fine to add.

Reply:
It's just that the current test can fail or pass for things unrelated to the functions we want to verify because it's non-specific.

I also think this really is a low hanging fruit. Check https://github.com/opendatakit/javarosa/commit/5eaee0782746b4fe502a89b5eac7e8b601621b43#diff-448a15bf3c816beba3aac62fe1c2fa24R195 to see what I mean (super straightforward behavior verification).

Reply:
In case this is just me being nitpicky about tests, and since @lognaturel is OK with it, I'll approve the PR

Reply:
Ok, please cherry pick that commit and merge. Your test would crash in the same way if there were a regression which is what's really important.